app/src/main/java/com/gustavoas/noti/services/AccessibilityService.kt

```diff
@@ -155,26 +161,145 @@ class AccessibilityService : AccessibilityService() {
         val paddingRight =
             (sharedPreferences.getInt("circularProgressBarMarginRight", 70) * 0.5).roundToInt() + 15
 
-        val param = circularProgressBar.layoutParams as LinearLayout.LayoutParams
-        param.setMargins(paddingLeft, paddingTop, paddingRight, 0)
-        circularProgressBar.layoutParams = param
+        // get layout params
+        val containerParams = container.layoutParams as ConstraintLayout.LayoutParams
+        val progressParams = circularProgressBar.layoutParams as FrameLayout.LayoutParams
+
+        // reset container constraints
+        containerParams.topToTop = -1
+        containerParams.bottomToBottom = -1
+        containerParams.leftToLeft = -1
+        containerParams.rightToRight = -1
+
+        // add navbar height to container margins
+        containerParams.setMargins(0, 0, 0, -navigationBarHeight)
```

Comment:
If it's only using the display over other apps permission and not the accessibility service it won't be able to get the navigationBarHeight here before the view is attached

```suggestion
        overlayView.doOnAttach {
                containerParams.setMargins(0, 0, 0, -navigationBarHeight)
        }
```

Reply:
> If it's only using the display over other apps permission and not the accessibility service it won't be able to get the navigationBarHeight here before the view is attached

that is correct. I'll add your suggestion


> This also seems to not be needed when using 3-button navigation

i was under the impression that the navigation bar with 3-button navigation will remain on the bottom edge of the screen, but at least on my pixel 6a this is not the case (the navigation bar remains on the side of the screen, where the charging port is). 

I'll add a check for this

Reply:
i've reworked the `navigationbarHeight` function and it should now handle 3-button navigation